### PR TITLE
SBC_COMMON : Redirect the user to create account if the user doesn't have valid role#3296

### PR DESCRIPTION
**Pull request by nitheesh-aot**
SBC_COMMON : Redirect the user to create account if the user doesn't have valid role#3296

### CODE REVIEW

File: vue/sbc-common-components/src/mixins/navigation-mixin.ts [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
     const resolvedRoutes = this.$router.resolve({ path: `/${routePath}` })
+    console.log(resolvedRoutes.resolved.matched)
     if (resolvedRoutes.resolved.matched.length > 0) {
       this.$router.push(`/${routePath}`)
     } else {
       window.location.assign(`${contextPath}${routePath}`)
     }
   }
+
+  protected redirectToPath (inAuth: boolean, routePath: string) {
```
Comment (sumesh-aot) [inline vue/sbc-common-components/src/mixins/navigation-mixin.ts:16]: navigateTo already handles route push if user is in auth page, else using window location. Do we still need this change here ?

File: vue/sbc-common-components/src/components/SbcSignin.vue [hunk 2]
```diff
@@ -36,34 +38,50 @@ import { KCUserProfile } from '../models/KCUserProfile'
...
+          const userInfo = await this.loadUserInfo()
           // sync the account if there is one
           await this.syncAccount()
-          this.$emit('sync-user-profile-ready')
           // eslint-disable-next-line no-console
           console.info('[SignIn.vue]Logged in User.Starting refreshTimer')
-          let tokenService = new TokenService()
-          await tokenService.init()
-          tokenService.scheduleRefreshTimer()
+          await KeyCloakService.initializeToken(this.$store)
```
Comment (sumesh-aot) [inline vue/sbc-common-components/src/components/SbcSignin.vue:64]: It looks like some unwanted calls here. There is initializeKeyCloak and then initializeToken. And initializeToken again does a keycloak init. Can we simplify this for better maintenance ?

File: vue/sbc-common-components/src/components/SbcSignin.vue [hunk 3]
```diff
@@ -36,34 +38,50 @@ import { KCUserProfile } from '../models/KCUserProfile'
...
+
+          // redirect to create account page if the user has no 'account holder' role
+          const isRedirectToCreateAccount = (userInfo.roles.includes(Role.PublicUser) && !userInfo.roles.includes(Role.AccountHolder))
```
[the remark this answers is not in the document]
Comment (sumesh-aot) [reply vue/sbc-common-components/src/components/SbcSignin.vue:67]: Account creation is a multi step process.. So we will not handle it now as I think there will be navigation option on the application to navigate to the app once account creation is done. For end user it's just **one** app, so we should be good.

File: vue/sbc-common-components/src/mixins/navigation-mixin.ts (hunk 1 above)
Comment (nitheesh-aot) [reply vue/sbc-common-components/src/mixins/navigation-mixin.ts:16]: You are right, will keep it as a single func

File: vue/sbc-common-components/src/components/SbcSignin.vue [hunk 4]
```diff
@@ -36,34 +38,50 @@ import { KCUserProfile } from '../models/KCUserProfile'
...
+          if (isRedirectToCreateAccount) {
+            switch (userInfo.loginSource) {
```
[the remark this answers is not in the document]
Comment (nitheesh-aot) [reply vue/sbc-common-components/src/components/SbcSignin.vue:68]: Yes, We can do that in the auth-web, but we need this to be here when signin come from the other applications.

[the remark this answers is not in the document]
Comment (saravanpa-aot) [reply vue/sbc-common-components/src/components/SbcSignin.vue:68]: hm.i was thinking of some solution where by we abstract this part to auth-web .. like a generic event or route in auth-web and auth-web handle the specificities of each login...
But this is good for now

**Pull request merged**